cdap-common/src/main/resources/cdap-default.xml

```diff
@@ -470,6 +470,24 @@
     </description>
   </property>
 
+  <property>
+    <name>run.record.counter.entry.retention.seconds</name>
+    <value>3600</value>
+    <description>
+      Maximum amount of time (in seconds) in which a run record counter entry is retained.
+      This is to safe guard launch requests flow-control such that if a request is somehow stuck in PENDING/STARTING
+      state, it will be dropped from the counter after retention period.
+    </description>
+  </property>
+
+  <property>
+    <name>run.record.counter.cleanup.interval.seconds</name>
+    <value>60</value>
```

Comment:
nit: 600. Seems fine to run very infrequent as your retention is 1hr.

Reply:
Changed it to 300.